extensions/rv32im/circuit/src/README.md

```diff
@@ -0,0 +1,250 @@
+# RV32IM Extension Circuit
+
+This directory contains the circuit implementation of the RV32IM extension.
+
+## Design
+
+The RV32IM chips consist of a series of an adapter chip and a core chip.
+
+- The adapter chip is responsible for adapting the input and output of the core chip to the format expected by the VM, and handling any interactions with the VM.
+- The core chip is responsible for implementing the logic of the RISC-V instructions.
+
+## Circuit statements
+
+This section describes the statements that each circuit is responsible for proving.
+Details about the constraints and assumptions for each statement are available in the implementation of the circuit.
+
+### Adapter
+
+#### 1. [ALU adapter](./adapters/alu.rs)
+
+Given `rs1`, `rs2`, and `rd` pointers, and a boolean indicating if `rs2` is an immediate value,
+this circuit proves the following:
+
+- A memory read from register `rs1` is performed
+- If `rs2` is not an immediate value, a memory read from register `rs2` is performed
+- A memory write to register `rd` is performed with the result of the operation
+- The instruction is executed successfully by the VM and the program counter is incremented by 4
```

Comment:
wording: the adapter never proves the instruction execution itself is executed successfully, what is proves is that the instruction was looked up correctly from the from_pc address in the program ROM, and that the to_pc after instruction execution is set correctly

can you reword all the "The instruction is executed succesfully [..]" phrases to something that more reflects the above?